assets/src/components/mapMarkers.tsx

```diff
@@ -217,6 +228,38 @@ const MobileFriendlyTooltip = ({
     </Popup>
   )
 }
+interface StopIconProps extends Omit<CircleMarkerProps, "center"> {
+  stop: Stop
+  radius?: number
+  selected?: boolean
+}
+
+export const StopIcon = ({
+  stop,
+  radius = 8,
+  selected = false,
+  ...props
+}: StopIconProps) => {
+  const ref = useRef<null | LeafletCircleMarker>(null)
+  const element = ref.current?.getElement()
+  useEffect(() => {
+    if (element) {
+      element.classList.toggle("selected", selected)
+    }
+  }, [selected, element])
```

Comment:
**question:** since `element` comes from a `ref`, that means changes to it won't necessarily trigger a re-render, correct, so this `useEffect` might not actually run when the `element` changes, right? Or am I missing something?

Reply:
Yeah you're right! 

I don't have a good understanding of leaflet's lifecycle and when `getElement()` can return `undefined`, and I was hoping the `useEffect` would help here, but it actually introduced a rendering error where it only respected `selected` on rerender.
I'm not quite sure what the right path forward is here, I've gotten it working a couple of different ways, but I've settled on a `useState` method that I'll have up in a bit for re-review.

Reply:
Please take a look at d7cbf91d87136cd2ba7ccb42a8a7434d23e4cf3c 🙏🏻

Reply:
I always forget that you can pass setters into `ref`! Yes, that looks good.